### ISSUE TITLE
Bug in no-unused-vars when using it in a string template (var is passed in a function param)

### ISSUE THREAD

**Issue by amalahussein** [eslint/eslint#4286]
We are using a variable from a function param in a string template, but eslint isnt recognizing it as being used.

Getting a no-unused-vars warning for the following code for 'verison':
```js
function stampBuild(version) {
    gulp.src('build/public/index.html')
        .pipe(insert.transform((contents) => {
            return contents.replace('</body>\n</html>', '<div>Build: ${version}</div>\n</body>\n</html>');
        }))
        .pipe(gulp.dest('build/public'));
}
```
Eslint verison:1.6.0
Console output:   28:21  warning  "version" is defined but never used  no-unused-vars

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Hmm.. I can't seem to be able to reproduce this with the following code:
```js
/* eslint no-unused-vars:2 */
var a = 5;
var b = 10;
console.log(`Fifteen is ${a + b} and\nnot ${2 * a + b}.`);
```

What version of NPM are you using? Can you double check version of escope package under `node_modules` in eslint installation directory?

**Comment by amalahussein** [comment]
Sure...checking now @ilyavolodin

**Comment by amalahussein** [comment]
{
  "name": "escope",
  "description": "ECMAScript scope analyzer",
  "homepage": "http://github.com/estools/escope",
  "main": "lib/index.js",
  "version": "3.2.0",
  "engines": {
    "node": ">=0.4.0"
  },

**Comment by mysticatea** [comment]
the example looks using single quotes. Templates are using backticks.

**Comment by amalahussein** [comment]
Yup that was it, it was a typo. :) Thanks so much guys, apology we didn't notice it earlier.
